packages/replay-next/src/hooks/useNotification.ts

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useEffect, useState } from "react";
+
+export type Permission = NotificationPermission | PermissionState;
+export type RequestPermission = () => Promise<boolean>;
+
+export function useNotification(): {
+  permission: Permission;
+  requested: boolean;
+  requestPermission: RequestPermission;
+  supported: boolean;
+} {
+  const [permission, setPermission] = useState<Permission>(Notification.permission);
```

Comment:
this is fine but since technically this is a mutable value maybe we should wrap this in a lazy init to make that more apparent?

```suggestion
  const [permission, setPermission] = useState<Permission>(() => Notification.permission);
```

Reply:
Using `useSyncExternalStore` will fix this.